packages/flame/lib/src/collisions/quadtree/broadphase.dart

```diff
@@ -0,0 +1,164 @@
+import 'dart:collection';
+
+import 'package:flame/collisions.dart';
+import 'package:flame/components.dart';
+import 'package:flame/extensions.dart';
+
+typedef ExternalBroadphaseCheck = bool Function(
+  PositionComponent one,
+  PositionComponent another,
+);
+
+typedef ExternalMinDistanceCheck = bool Function(
+  Vector2 activeItemCenter,
+  Vector2 potentialCenter,
+);
+
+/// Performs Quad Tree broadphase check
+/// See [HasQuadTreeCollisionDetection.initCollisionDetection] for detail
+/// description of its parameters
+class QuadTreeBroadphase<T extends Hitbox<T>> extends Broadphase<T> {
+  QuadTreeBroadphase({
+    super.items,
+    required Rect mainBoxSize,
```

Comment:
Maybe this should be an `Aabb2` since that is what we are using everywhere else, or is `Rect` more efficient here?

Reply:
I cant to predict the best solution for end-user, but can confidently say that it would not be Aabb2 class. 
Map dimensions can be get from `TiledComponent` if map loaded from file. For generated map it could be pure width and height.  
It seems to me that Rect is just more obviouse here: it represents an area where collision would be calculated. 
But maybe it's only my opinion that it would be user-friendly :-)

Reply:
Since the `TiledComponent` soon will be a `PositonComponent` and can therefore also be rotated it is a lot easier to handle rotations in `Aabb2` since the bounding box needs to grow when that happens. And the `PositionComponent` already exposes an Aabb2 if I remember correctly, or maybe the rect that it exposes already takes the angle into consideration?

Reply:
Nvm, the `toRect` gives back the axis-aligned rect.

Reply:
To be honest, I'd prefer not to handle rotations of whole map in this algoritm and just to make bigger "bounding box' instead...

Reply:
Makes sense, so the user just have to send in a larger rect if they plan to rotate the component.